File: backend/src/main/java/fr/daddyornot/universdesames/model/Product.java
package fr.daddyornot.universdesames.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@Table(name = "products")
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private Double price;

    // Remplacement de ElementCollection par ManyToMany
    @ManyToMany
    @JoinTable(
        name = "product_stones_link",
        joinColumns = @JoinColumn(name = "product_id"),
        inverseJoinColumns = @JoinColumn(name = "stone_id")
    )
    private List<Stone> stones = new ArrayList<>();

    @Column(name = "image_url")
    private String imageUrl;

    @Enumerated(EnumType.STRING)
    private ProductType type;

    private Integer sessionCount;
    private Integer durationMonths;
    
    private Integer bufferTimeMinutes = 0;

    // --- ABONNEMENT ---
    private boolean isSubscription = false;
    
    // "month", "year", "week", "day"
    private String recurringInterval;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<ProductVariant> variants = new ArrayList<>();

    @ManyToMany
    @JoinTable(
        name = "product_available_sizes",
        joinColumns = @JoinColumn(name = "product_id"),
        inverseJoinColumns = @JoinColumn(name = "size_id")
    )
    private List<ProductSize> sizes = new ArrayList<>();
}

File: frontend/src/app/core/models/product.ts
import { Stone } from './stone';

export type ProductType = 'PHYSICAL' | 'ENERGY_CARE' | 'CARD_READING' | 'COACHING' | 'SUBSCRIPTION';

export interface ProductVariant {
  id?: number;
  label: string;
  price: number;
  sessionCount?: number;
  durationMonths?: number;
}

export interface ProductSize {
  id: number;
  label: string;
  description: string;
}

export interface Product {
  id: number;
  name: string;
  description: string;
  price: number;
  stones: Stone[];
  imageUrl: string;
  type: ProductType;
  sessionCount?: number;
  durationMonths?: number;
  variants?: ProductVariant[];
  sizes?: ProductSize[];
  bufferTimeMinutes?: number;
  isSubscription?: boolean;
  recurringInterval?: string;
}
